requirements-dev.txt

```diff
@@ -20,7 +22,9 @@ pillow
 pycodestyle
 pytest
 selenium
+scipy
```

Comment:
Same for scipy

Reply:
tl;dr it;s on cartopy not handling dependencies properly, but essentially it breaks
TestNotebooks.test_geodedetic_image_overlay



<details>
<summary> full traceback </summary>

~~~
______________________________________ TestNotebooks.test_geodedetic_image_overlay ______________________________________

self = <test_notebooks.NotebookTester object at 0x7f13471bfd30>, exporter = None, filename = None

    def __call__(self, exporter=None, filename=None):
        raw_nb = nbconvert.exporters.Exporter().from_filename(self.filename)
        raw_nb[0].metadata.setdefault('kernelspec', {})['name'] = 'python'
>       exec_nb = nbconvert.preprocessors.ExecutePreprocessor(timeout=600).preprocess(*raw_nb)

tests/notebooks/test_notebooks.py:29: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
venv/lib/python3.5/site-packages/nbconvert/preprocessors/execute.py:262: in preprocess
    nb, resources = super(ExecutePreprocessor, self).preprocess(nb, resources)
venv/lib/python3.5/site-packages/nbconvert/preprocessors/base.py:69: in preprocess
    nb.cells[index], resources = self.preprocess_cell(cell, resources, index)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <nbconvert.preprocessors.execute.ExecutePreprocessor object at 0x7f134398beb8>
cell = {'execution_count': 6, 'source': "import cartopy.crs as ccrs\nfrom cartopy.img_transform import warp_array\n\nsource_e... 'scipy'"], 'ename': 'ImportError', 'output_type': 'error', 'evalue': "No module named 'scipy'"}], 'cell_type': 'code'}
resources = defaultdict(None, {'metadata': defaultdict(None, {'name': 'GeodedeticImageOverlay', 'path': '/home/demetrio/Parkbob/git/folium/tests/notebooks/../../examples', 'modified_date': 'September 3, 2018'}), 'output_extension': '.txt'})
cell_index = 8

    def preprocess_cell(self, cell, resources, cell_index):
        """
            Executes a single code cell. See base.py for details.
    
            To execute all cells see :meth:`preprocess`.
            """
        if cell.cell_type != 'code':
            return cell, resources
    
        reply, outputs = self.run_cell(cell, cell_index)
        cell.outputs = outputs
    
        if not self.allow_errors:
            for out in outputs:
                if out.output_type == 'error':
>                   raise CellExecutionError.from_cell_and_msg(cell, out)
E                   nbconvert.preprocessors.execute.CellExecutionError: An error occurred while executing the following cell:
E                   ------------------
E                   import cartopy.crs as ccrs
E                   from cartopy.img_transform import warp_array
E                   
E                   source_extent = [lon.min(), lon.max(), lat.min(), lat.max()]
E                   
E                   new_data = warp_array(colored_data,
E                                         target_proj=ccrs.GOOGLE_MERCATOR,
E                                         source_proj=ccrs.PlateCarree(),
E                                         target_res=data.shape,
E                                         source_extent=source_extent,
E                                         target_extent=None,
E                                         mask_extrapolated=False)
E                   
E                   
E                   m = folium.Map(location=[lat.mean(), lon.mean()], zoom_start=1)
E                   
E                   folium.raster_layers.ImageOverlay(
E                       image=new_data[0],
E                       bounds=[[lat.min(), lon.min()], [lat.max(), lon.max()]],
E                       opacity=0.25
E                   ).add_to(m)
E                   
E                   m.save(os.path.join('results', 'GeodedeticImageOverlay_2.html'))
E                   
E                   m
E                   ------------------
E                   
E                   ---------------------------------------------------------------------------
E                   ImportError                               Traceback (most recent call last)
E                   <ipython-input-6-f0b9096a59bd> in <module>()
E                         1 import cartopy.crs as ccrs
E                   ----> 2 from cartopy.img_transform import warp_array
E                         3 
E                         4 source_extent = [lon.min(), lon.max(), lat.min(), lat.max()]
E                         5 
E                   
E                   ~/Parkbob/git/folium/venv/lib/python3.5/site-packages/cartopy/img_transform.py in <module>()
E                        24 
E                        25 import numpy as np
E                   ---> 26 import scipy.spatial
E                        27 
E                        28 import cartopy.crs as ccrs
E                   
E                   ImportError: No module named 'scipy'
E                   ImportError: No module named 'scipy'

venv/lib/python3.5/site-packages/nbconvert/preprocessors/execute.py:286: CellExecutionError
~~~
</details>

they already know that: https://github.com/SciTools/cartopy/issues/1112